### PR TITLE
Add assertion and move method FindCommandParser and DeleteCommandParser methods for ranges into ParserUtil

### PR DESCRIPTION
**Pull request by Tohtoroo**

### CODE REVIEW

File: src/main/java/seedu/dengue/model/predicate/DeleteDatePredicate.java [hunk 1]
```diff
@@ -32,7 +33,7 @@ public DeleteDatePredicate(Optional<Date> date, Range<Date> range) {
      * @param date (Optional) Date input passed in by user.
      */
```
Comment (axmszr) [inline src/main/java/seedu/dengue/model/predicate/DeleteDatePredicate.java:35]: Extra line! I think it won't mess up the JavaDoc but just to be safe?

File: src/main/java/seedu/dengue/model/predicate/FindPredicate.java [hunk 2]
```diff
@@ -45,8 +45,24 @@ public FindPredicate(Optional<Name> name, Optional<SubPostal> subPostal, Optiona
         this.dateRange = dateRange;
         this.ageRange = ageRange;
     }
+
+    private boolean isEveryPrefixEmpty(Optional<Name> name, Optional<SubPostal> subPostal,
```
Comment (axmszr) [inline src/main/java/seedu/dengue/model/predicate/FindPredicate.java:49]: Maybe name it something like `allPrefixesEmpty`? So it sounds more like a boolean condition than a question.

File: src/main/java/seedu/dengue/model/predicate/PredicateUtil.java [hunk 3]
```diff
@@ -76,6 +77,38 @@ public static boolean isVariantValid(String variant) {
...
 
+    /**
+     * Checks the validity of the date range
+     * @param start the start
+     * @param end the end
+     * @return true if the start date begins after the end date, false otherwise.
+     */
+    public static boolean isDateRangeValid(Date start, Date end) {
+        assert start != null;
+        assert end != null;
```
Comment (axmszr) [inline src/main/java/seedu/dengue/model/predicate/PredicateUtil.java:88]: I think `requireNonNull` covers this already.

```suggestion
```

File: src/main/java/seedu/dengue/model/predicate/PredicateUtil.java [hunk 4]
```diff
@@ -76,6 +77,38 @@ public static boolean isVariantValid(String variant) {
...
+
+    /**
+     * Checks the validity of the age range
+     * @param start the start
+     * @param end the end
+     * @return true if the start age begins after the end age, false otherwise.
+     */
+    public static boolean isAgeRangeValid(Age start, Age end) {
+        assert start != null;
+        assert end != null;
```
Comment (axmszr) [inline src/main/java/seedu/dengue/model/predicate/PredicateUtil.java:104]: Same as above!

```suggestion
```

**Review by axmszr — CHANGES_REQUESTED** [review]
Couple nits, but otherwise LGTM!

File: src/main/java/seedu/dengue/model/predicate/FindPredicate.java (hunk 2 above)
Comment (Tohtoroo) [reply src/main/java/seedu/dengue/model/predicate/FindPredicate.java:49]: image
Should be can to leave it as is for a boolean method

File: src/main/java/seedu/dengue/model/predicate/DeleteDatePredicate.java (hunk 1 above)
Comment (axmszr) [inline src/main/java/seedu/dengue/model/predicate/DeleteDatePredicate.java:35]: Deletomundo

```suggestion
```

**Review by axmszr — CHANGES_REQUESTED** [review]
Added as a suggestion!

**Review by axmszr — APPROVED** [review]

**Pull request merged**